### PR TITLE
29852 Remove appointment list facilities request 2

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Description
This PR removes the extra api call to retreive facility information since that information is now included in the appointments api call.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#29852

## Testing done
Unit testing

## Screenshots
N/A

## Acceptance criteria
- [ ] All test should pass

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/appointment-list/redux/actions.js [hunk 1]
```diff
@@ -150,6 +151,25 @@ async function getAdditionalFacilityInfo(futureAppointments, useV2 = false) {
...
+  // Facility information included with v2 appointment api call.
+  const facilities = appointments
+    ?.map(
+      appt =>
+        appt.vaos.apiData.location
+          ? { ...appt.vaos.apiData.location.attributes }
+          : null,
+    )
+    .filter(n => n);
+  return facilities ? transformFacilitiesV2(facilities) : [];
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/redux/actions.js:170]: It seems like this whole function could be `return appointments.map(appt => appt.vaos.facilityData).filter(n => n)`. I don't think you need the transform here, since it's in the appointment service.

File: src/applications/vaos/appointment-list/redux/actions.js [hunk 2]
```diff
@@ -270,12 +290,20 @@ export function fetchFutureAppointments({ includeRequests = true } = {}) {
...
 
       try {
-        const facilityData = await getAdditionalFacilityInfo(
-          [].concat(...data),
-          featureFacilitiesServiceV2,
-        );
+        let facilityData;
+        if (
+          featureVAOSServiceVAAppointments ||
+          featureVAOSServiceCCAppointments
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/redux/actions.js:296]: I think you can leave off checking `featureVAOSServiceCCAppointments`, we don't use facility ids from those appointments.

File: src/applications/vaos/appointment-list/redux/actions.js [hunk 3]
```diff
@@ -414,12 +442,22 @@ export function fetchPastAppointments(startDate, endDate, selectedIndex) {
...
 
       try {
-        const facilityData = await getAdditionalFacilityInfo(
-          getState().appointments.past,
-          featureFacilitiesServiceV2,
-        );
+        let facilityData;
+        if (
+          featureVAOSServiceVAAppointments ||
+          featureVAOSServiceCCAppointments
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/redux/actions.js:448]: Same comment as above, I think you can skip checking `featureVAOSServiceCCAppointments`.

File: src/applications/vaos/appointment-list/redux/actions.js [hunk 4]
```diff
@@ -414,12 +442,22 @@ export function fetchPastAppointments(startDate, endDate, selectedIndex) {
       });
 
       try {
-        const facilityData = await getAdditionalFacilityInfo(
-          getState().appointments.past,
-          featureFacilitiesServiceV2,
-        );
+        let facilityData;
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/redux/actions.js:445]: I think we also want to make these changes in `fetchPendingAppointments`. Same updates as you've made for the other two places, just checking `featureVAOSServiceRequests` instead of `featureVAOSServiceVAAppointments`.

We could leave this for a separate PR, too, though.

File: src/applications/vaos/appointment-list/redux/actions.js (hunk 2 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/redux/actions.js:296]: Done

File: src/applications/vaos/appointment-list/redux/actions.js (hunk 4 above)
Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/redux/actions.js:445]: Yea, I actually removed it because it seemed like the requests would have already been retrieved from the fetchFutureAppointments call. But I see that 'includeRequests' flag is false.

Comment (vbahinwillit) [reply src/applications/vaos/appointment-list/redux/actions.js:445]: Done

**Comment by jbalboni** [comment]
It looks like locally if I open the appointment list and choose a VA appointment, I see just the facility name and not the address on the details page. If I refresh the page, everything shows up.

**Review by jbalboni — APPROVED** [review]
